### PR TITLE
NECO-185: Added meta APIs on README

### PR DESCRIPTION
**Pull request by YukiThornton**
## Ticket
- https://pflabo.atlassian.net/browse/NECO-185
## Changes
- README
  - Added meta APIs
  - Added Requirements and References
  - Deleted License
  - Fixed some sentences.
- Source code
  - Added empty methods (addMeta, removeMeta)
  - Modified meta method.
## Tests
- See the ticket.
## How It Looks Like
- https://github.com/okunoyuki/media-storage-rb/tree/feature/NECO-185

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,47 +23,127 @@ And then execute:
...
+### Obtain an access token
+
+```ruby
+client = RicohAPI::Auth::Client.new(
+  '<your-client-id>',
+  '<your-client-secret>'
+)
+client.resource_owner_credentials = '<your-user-id>', '<your-password>'
+api_session = client.api_token_for! RicohAPI::MStorage::SCOPE
+your_access_token = api_session.access_token
```
Comment (sohda) [inline README.md]: This local variable should be `access_token` .

File: README.md [hunk 2]
```diff
@@ -14,47 +23,127 @@ And then execute:
...
+```ruby
 mstorage.list limit: 25
 mstorage.list after: '<cursor-id>'
+```
+
+* Search
+
+Also, you can get a list searched by user metadata.
+```ruby
+filter = {'meta.user.<key1>'=>'<value1>', 'meta.user.<key2>'=>'<value2>'}
```
Comment (sohda) [inline README.md]: You should put white space before and after the operator: `=>` .
- also: https://github.com/sohda/media-storage-rb/pull/1/files#diff-04c6e90faac2675aa89e2176d2eec7d8R108

File: README.md [hunk 3]
```diff
@@ -14,47 +23,127 @@ And then execute:
...
+* User metadata (with a key)
+```ruby
+mstorage.meta '<media-id>', 'user.<key>'
+```
 
-The gem is available as open source under the terms of the [MIT License](http://opensource.org/licenses/MIT).
+### Delete media metadata
+* User metadata (all)
+```ruby
+mstorage.removeMeta '<media-id>', 'user'
```
Comment (sohda) [inline README.md]: You should `mstorage.remove_meta ....` .
You should use the snake case for method name and variable name.
- also:
  - https://github.com/sohda/media-storage-rb/pull/1/files#diff-04c6e90faac2675aa89e2176d2eec7d8R145
  - https://github.com/sohda/media-storage-rb/pull/1/files#diff-fdae4052fdc7c416626ef650ac8dced9R42
  - https://github.com/sohda/media-storage-rb/pull/1/files#diff-fdae4052fdc7c416626ef650ac8dced9R63
  - https://github.com/sohda/media-storage-rb/pull/1/files#diff-fdae4052fdc7c416626ef650ac8dced9R68

File: README.md (hunk 1 above)
Comment (YukiThornton) [inline README.md]: Okay, I'll fix it soon.

File: README.md (hunk 2 above)
Comment (YukiThornton) [inline README.md]: Okay, I'll fix it soon.

File: README.md (hunk 3 above)
Comment (YukiThornton) [inline README.md]: Thank you! I'm gonna fix it right away.

File: README.md [hunk 4]
```diff
@@ -14,47 +23,127 @@ And then execute:
...
+## SDK API
 
-To obtain access tokens, you can use [ricohapi-auth gem](https://github.com/ricohapi/auth-rb).
+### Obtain an access token
+
+```ruby
+client = RicohAPI::Auth::Client.new(
+  '<your-client-id>',
+  '<your-client-secret>'
+)
```
Comment (sohda) [inline README.md]: I think the following code is more Ruby-ish `lient = RicohAPI::Auth::Client.new '<your-client-id>', '<your-client-secret>'` than above code.
Also: https://github.com/sohda/media-storage-rb/pull/1/files#diff-04c6e90faac2675aa89e2176d2eec7d8R51

File: README.md [hunk 5]
```diff
@@ -14,47 +23,127 @@ And then execute:
...
 
 You can find [a working rails sample app using Ricoh Media Storage API](https://github.com/ricohapi/media-storage-sample-app).
 
-### Obtaining Access Token for Media Storage API
+## SDK API
 
-To obtain access tokens, you can use [ricohapi-auth gem](https://github.com/ricohapi/auth-rb).
+### Obtain an access token
+
+```ruby
```
Comment (sohda) [inline README.md:39]: I think it is better to add `require 'ricohapi/mstorage'` .

File: README.md (hunk 4 above)
Comment (YukiThornton) [inline README.md]: I'll fix it.

File: README.md (hunk 5 above)
Comment (YukiThornton) [inline README.md:39]: I added Sample Flow section and to show how to require the gem.

File: README.md [hunk 6]
```diff
@@ -14,47 +23,123 @@ And then execute:
...
 
     $ gem install ricohapi-mstorage
 
 ## Usage
 
 You can find [a working rails sample app using Ricoh Media Storage API](https://github.com/ricohapi/media-storage-sample-app).
 
-### Obtaining Access Token for Media Storage API
+## SDK API
+Require `ricohapi/mstorage` to use the APIs below.
```
Comment (sohda) [inline README.md]: I think the following explanation is better: "Add `require 'ricohapi/mstorage'` to use the APIs below."
Because end user can copy it.

Comment (YukiThornton) [inline README.md]: Thank you! I'm gonna fix it soon.

**Comment by sohda** [comment]
I reviewed all the changes.
I'm going to merge this pull request.

**Pull request merged**